### ISSUE TITLE
reproducing zero-shot results of visual dialog on instructblip

### ISSUE THREAD

**Issue by gyhdog99** [salesforce/LAVIS#439]
Hi, 

I am currently trying to reproducing the zero-shot evaluation results of instructblip on visual dialog.
However, I can only get a mrr of around 38. Here are my implementation details:

data:
VisDial dataset v1.0, validation set, downloaded from https://visualdialog.org/data, Dialogs (17M), Images (331M)

input:
there are 2064 dialogs, each with 10 rounds, I use the first 9 rounds as the dialog history, the last one as the actual question, here is an example of my inputs:
'This image has the caption "a bedroom is filled with lots of posters and a busy computer desk".Dialog history: Question: is the photo in color? Answer: yes. Question: is it a professional photo? Answer: no. Question: is it well lit? Answer: no. Question: is it daytime? Answer: i don\'t see windows. Question: does this look like an adults bedroom? Answer: maybe. Question: is the room clean? Answer: no. Question: can you tell what kind of computer it is? Answer: no not really. Question: is it a flat screen? Answer: yes. Question: what\'s the desk made out of? Answer: cheap plastic or wood.\nQuestion: is there a computer chair? Short answer: '

evaluation:
blip2_t5_instruct_xl is adopted, I use self.predict_class() to do the inference, answer candidates are from the options of the last round. 
The MRR metric is computed as follows:
`        # preds: [bs, num_options], returned by the method model.predict_class(), option index sorted by the loss in a ascending way.
        # targets: [bs], the index of the ground truth answer in the options.
        targets_ = torch.tensor(targets)
        targets_ = targets_.unsqueeze(1).repeat(1, preds.size(1))
        ranks = torch.argwhere(targets_ == preds)[:, 1].numpy()
        ranks = ranks + 1
        mrr = np.reciprocal(ranks).mean()`

Doing this way, I can only get a MRR of 38 instead of 46.6 as reported in the paper. I would be very appreciative if you can provide me with more details about this!
